test/test_analyse.ml

```diff
@@ -0,0 +1,114 @@
+open Lwt.Infix
+
+module Analysis = struct
+  include Ocaml_ci.Analyse.Analysis
+
+  let set_equality = Alcotest.(equal (slist string String.compare))
+
+  (* Make the [t] type concrete from the observable fields for easier testing *)
+  type t = {
+    opam_files : string list; [@equal set_equality]
```

Comment:
I disagree; the fact that the output is sorted is an undocumented implementation detail that I don't think should be relied upon by the tests (and is not very useful anyway).

Reply:
The fact that the output is sorted is an implementation detail but I think the fact that the output is always in the same order is important. That could be used as a key in some caches.

Reply:
As discussed in person, I'm going to stick with my guns on this one and not test undocumented behaviours. I had a look in the codebase, and I don't see any occurrence of caching `Analysis.t` values (this would be odd, since `Analysis.t` is abstract). If you want, we can alter the `Analysis.t` type in a later PR :slightly_smiling_face:

Reply:
I agree :)
My point on caching is that maybe in the future, the list of opam files (not `Analyse.t`) or a list generated from it may be used as a key. It's not important now and the implementation does the right thing.